src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -326,163 +983,392 @@ private void onNetwork(
+
+        private void doNetworkReset(
+            long traceId,
+            long authorization)
+        {
+            cleanupDecodeSlotIfNecessary();
+            doReset(network, routeId, initialId, traceId, authorization, EMPTY_OCTETS);
+        }
+
+        private void doNetworkWindow(
+            long traceId,
+            long authorization,
+            int credit,
+            int padding,
+            long budgetId)
+        {
+            assert credit > 0;
+
+            initialBudget += credit;
+            doWindow(network, routeId, initialId, traceId, authorization, budgetId, credit, padding);
+        }
+
+        private void doNetworkSignal(
+            long traceId)
+        {
+            doSignal(network, routeId, initialId, traceId);
+        }
+
+        private void decodeNetwork(
+            long traceId,
+            long authorization,
+            long budgetId,
+            DirectBuffer buffer,
+            int offset,
+            int limit)
+        {
+            AmqpServerDecoder previous = null;
+            int progress = offset;
+            while (progress <= limit && previous != decoder)
+            {
+                previous = decoder;
+                progress = decoder.decode(this, traceId, authorization, budgetId, buffer, progress, limit);
+            }
+
+            if (progress < limit)
+            {
+                if (decodeSlot == NO_SLOT)
                 {
-                    decodeState = this::decodeFrame;
+                    decodeSlot = bufferPool.acquire(initialId);
+                }
+
+                if (decodeSlot == NO_SLOT)
+                {
+                    cleanupNetwork(traceId, authorization);
                 }
                 else
                 {
-                    decodeState = this::decodeInvalid;
-                    doEnd();
+                    final MutableDirectBuffer slotBuffer = bufferPool.buffer(decodeSlot);
+                    decodeSlotLimit = limit - progress;
+                    slotBuffer.putBytes(0, buffer, progress, decodeSlotLimit);
                 }
             }
+            else
+            {
+                cleanupDecodeSlotIfNecessary();
+            }
+        }
+
+        private void onDecodeHeader(
+            long traceId,
+            long authorization,
+            AmqpProtocolHeaderFW header)
+        {
+            if (isAmqpHeaderValid(header))
+            {
+                doEncodeHeader(header.major(), header.minor(), header.revision(), traceId, authorization);
+            }
+        }
+
+        private void onDecodeFrameHeader(
+            AmqpFrameHeaderFW frameHeader)
+        {
+            doEncodeFrameHeader(frameHeader.doff(), frameHeader.type(), frameHeader.channel(), frameHeader.performative());
+        }
+
+        private void onDecodeOpen(
+            long traceId,
+            long authorization,
+            AmqpOpenFW open)
+        {
+            containerId = open.containerId().asString();
```

Comment:
Do we still need to retain this?
I think we were going to remove `containerId` from `AmqpBeginEx`.

Reply:
We do not need to retain this `containerId` anymore. I removed `containerId` from `AmqpBeginEx` in `nukleus-amqp.spec` and `command-log`.